src/build_upgrade.rs

```diff
@@ -463,179 +488,110 @@ fn generate_authorize_upgrade_calls(upgrade_details: &UpgradeDetails) -> Vec<Cal
 	authorization_calls
 }
 
-// Generate the `system.set_code` call that will upgrade the Relay Chain.
-fn generate_relay_upgrade_call(upgrade_details: &UpgradeDetails) -> Option<CallInfo> {
-	println!("\nGenerating Relay Chain upgrade call. The runtime hash is logged if you would like to verify it with srtool.\n");
-	// None if there is no version.
-	upgrade_details.relay_version.clone()?;
-	let runtime_version = semver_to_intver(&upgrade_details.relay_version.clone().unwrap());
-	match upgrade_details.relay {
-		Network::Kusama => {
-			use kusama_relay::runtime_types::frame_system::pallet::Call as SystemCall;
-
-			let path = format!(
-				"{}kusama_runtime-v{}.compact.compressed.wasm",
-				upgrade_details.directory, runtime_version
-			);
-			let runtime = fs::read(path).expect("Should give a valid file path");
-			let runtime_hash = blake2_256(&runtime);
-			println!("Kusama Relay Chain Runtime Hash: 0x{}", hex::encode(runtime_hash));
-
-			if !upgrade_details.set_relay_directly {
-				// authorize upgrade
-				Some(CallInfo::from_runtime_call(NetworkRuntimeCall::Kusama(
-					KusamaRuntimeCall::System(SystemCall::authorize_upgrade {
-						code_hash: H256(runtime_hash),
-					}),
-				)))
-			} else {
-				// set code directly
-				Some(CallInfo::from_runtime_call(NetworkRuntimeCall::Kusama(
-					KusamaRuntimeCall::System(SystemCall::set_code { code: runtime }),
-				)))
-			}
-		},
-		Network::Polkadot => {
-			use polkadot_relay::runtime_types::frame_system::pallet::Call as SystemCall;
-
-			let path = format!(
-				"{}polkadot_runtime-v{}.compact.compressed.wasm",
-				upgrade_details.directory, runtime_version
-			);
-			let runtime = fs::read(path).expect("Should give a valid file path");
-			let runtime_hash = blake2_256(&runtime);
-			println!("Polkadot Relay Chain Runtime Hash: 0x{}", hex::encode(runtime_hash));
-
-			if !upgrade_details.set_relay_directly {
-				// authorize upgrade
-				Some(CallInfo::from_runtime_call(NetworkRuntimeCall::Polkadot(
-					PolkadotRuntimeCall::System(SystemCall::authorize_upgrade {
-						code_hash: H256(runtime_hash),
-					}),
-				)))
-			} else {
-				// set code directly
-				Some(CallInfo::from_runtime_call(NetworkRuntimeCall::Polkadot(
-					PolkadotRuntimeCall::System(SystemCall::set_code { code: runtime }),
-				)))
-			}
-		},
-		_ => panic!("Not a Relay Chain"),
-	}
-}
-
 // Take the parachain authorization calls and the Relay Chain call, and batch them into one call
 // that can be executed on the Relay Chain. The call returned here is the proposal to put to
 // referendum.
-async fn construct_batch(
-	upgrade_details: &UpgradeDetails,
-	relay_call: Option<CallInfo>,
-	para_calls: Vec<CallInfo>,
-) -> CallInfo {
+async fn construct_batch(upgrade_details: &UpgradeDetails, para_calls: Vec<CallInfo>) -> CallInfo {
```

Comment:
Could just be `calls` now
```suggestion
async fn construct_batch(upgrade_details: &UpgradeDetails, calls: Vec<CallInfo>) -> CallInfo {
```